### ISSUE TITLE
New fixed point substitution breaks  forward model practice possibly copied by users from hm tutorial

### ISSUE THREAD

**Issue by eivindjahren** [equinor/ert#4473]
In the hm tutorial, the RMS forward model would be called like this:

```
FORWARD_MODEL RMS(<IENS>=<IENS>...)
```

As we do substitutions now, the parameter gets put in a context where IENS is not defined yet, so it is not replaced.

Likely we would want to remedy the situation with two steps:

- [ ] Change substitute_real_iter to make a copy and then appends the per real/iter defines. This reduces the chance that fixed point replace fails. https://github.com/equinor/ert/blob/365be9f9891606af50d85c1aa82a6938e9c18f8b/src/ert/_c_wrappers/util/substitution_list.py#L117
- [ ] Swap the order of private arg and context substitutions in the job printing, and call substitute_real_iter for both https://github.com/equinor/ert/blob/365be9f9891606af50d85c1aa82a6938e9c18f8b/src/ert/_c_wrappers/job_queue/forward_model.py#L59-L61 . This reduces the chance that fixed point substitute fails. One can also create a copy of context, add the private defines to it and then call substitute which further minimizes problems.
- [ ] Provide a deprecation warning whenever a substitution of the form `<ITER>=<ITER>` is added to the substitution list.

**Comment by oysteoh** [comment]
Would `Provide a deprecation warning whenever a substitution of the form <ITER>=<ITER> is added to the substitution list.` automatically add this "problem" to the new feature we are providing suggesting how to fix? If not - i think this issue is also something they should be informed about and a suggestion on how to remove the warning 👍

**Comment by eivindjahren** [comment]
Yes, some suggestion would indeed be helpful. The whole substitutions thing has a lot of caveats so we need to be careful.
